src/Chirp.Razor/Program.cs

```diff
@@ -1,49 +1,81 @@
 using Microsoft.EntityFrameworkCore;
-using Microsoft.AspNetCore.Authentication;
+using Chirp.Core;
 using Microsoft.AspNetCore.Authentication.OpenIdConnect;
-using Microsoft.AspNetCore.Authorization;
 using Microsoft.Identity.Web;
 using Microsoft.Identity.Web.UI;
 using Microsoft.IdentityModel.Logging;
-using Microsoft.AspNetCore.Authentication.JwtBearer;
+using FluentValidation;
+using Microsoft.Data.SqlClient;
+using Microsoft.EntityFrameworkCore.Migrations;
 
 var builder = WebApplication.CreateBuilder(args);
 
-builder.Environment.EnvironmentName = "Development";
 
-var connectionString = $"Data source={Path.Combine(Path.GetTempPath() + "chirp.db")}";
-builder.Services.AddDbContext<ChirpDBContext>(options => options.UseSqlite(connectionString));
+// Add services to the container.
+builder.Services.AddRazorPages();
+
+builder.Configuration.AddJsonFile("appSettings.json", optional: false, reloadOnChange: true).AddJsonFile($"appSettings.{builder.Environment.EnvironmentName}.json", optional: true);
+builder.Environment.EnvironmentName = "Development";
 
-builder.Services.AddScoped<AuthorRepository>();
-builder.Services.AddScoped<CheepRepository>();
-builder.Configuration.AddJsonFile("appSettings.json", optional: false, reloadOnChange: true)
-    .AddJsonFile($"appSettings.{builder.Environment.EnvironmentName}.json", optional: true);
+SqlConnectionStringBuilder stringBuilder = new SqlConnectionStringBuilder();
+stringBuilder.DataSource = "bdsagroup4-chirpdb.database.windows.net";
+stringBuilder.UserID = "azureuser";
+stringBuilder.Password = "Ab12345_";
+stringBuilder.InitialCatalog = "bdsagroup4-chirpdb";
 
 
-using (var context = new ChirpDBContext())
+if (builder.Environment.IsDevelopment())
 {
-    context.Database.EnsureCreated();
-    DbInitializer.SeedDatabase(context);
+    builder.Services.AddDbContext<ChirpDBContext>(options =>
+    {
+        options.UseSqlServer(builder.Configuration.GetConnectionString("ChirpDB"));
+    });
+}
+else
+{
+    builder.Services.AddDbContext<ChirpDBContext>(options =>
+    {
+        options.UseSqlServer(stringBuilder.ConnectionString);
+    });
 }
 
+builder.Services.AddScoped<AbstractValidator<CheepCreateDTO>, CheepCreateValidator>();
+builder.Services.AddScoped<IAuthorRepository, AuthorRepository>();
+builder.Services.AddScoped<ICheepRepository, CheepRepository>();
+builder.Services.AddScoped<ICheepService, CheepService>();
+
 builder.Services.AddAuthentication(OpenIdConnectDefaults.AuthenticationScheme)
     .AddMicrosoftIdentityWebApp(builder.Configuration.GetSection("AzureAdB2C"));
 builder.Services.AddRazorPages()
     .AddMicrosoftIdentityUI();
 
 var app = builder.Build();
 
+
+
+using var scope = app.Services.CreateScope();
+var services = scope.ServiceProvider;
+var dbContext = services.GetRequiredService<ChirpDBContext>();
+
+if (dbContext.Database.EnsureDeleted())
+{
+    dbContext.Database.Migrate();
+    DbInitializer.SeedDatabase(dbContext);
+}
+
+// Configure the HTTP request pipeline.
```

Comment:
Perphaps we should add some comments on why theese are necessary / what they include (on a high level).
Since it is a large task, perphaps we should create an issue regarding this? :)